src/commands/language.ts

```diff
@@ -0,0 +1,82 @@
+import { botCache } from "../../deps.ts";
+import { PermissionLevels } from "../types/commands.ts";
+import { createSubcommand, createCommand, getCurrentLanguage } from "../utils/helpers.ts";
+import { Embed } from "../utils/Embed.ts";
+import { db } from "../database/database.ts";
+
+const allowedLanguages = [
+  { id: "en_US", flag: ":flag_us:", name: "English"},
+  { id: "cs_CZ", flag:":flag_cz:", name: "Czech"}
+];
+
+createCommand({
+  name: "language",
+  aliases: ['lang'],
+  arguments: [
+    {
+      name: "subcommand",
+      type: "subcommand",
+      required: false,
+    },
+  ],
+  guildOnly: true,
+  permissionLevels: [PermissionLevels.MEMBER],
+  execute: async function (message) {
+    const currentLanguageId = getCurrentLanguage(message.guildID);
+    let currentLanguage = allowedLanguages.find(item => item.id === currentLanguageId) || allowedLanguages[0];
+    const embed = new Embed()
+      .setTitle("Language Information")
+      .setDescription(`**Current Language**: ${currentLanguage.flag} - \`${currentLanguage.name}\``)
+      .setTimestamp();
+
+    message.send({ embed });
+  },
+});
+
+createSubcommand("language", {
+  name: "set",
+  arguments: [
+    {
+      name: "language",
+      type: "string",
+      required: true,
+      missing: async function(message) {
+        const listOfLanguages = allowedLanguages.map(lang => `${lang.flag} - \`${lang.name}\``).join('\n');
+        const embed = new Embed()
+          .setTitle("Available Languages")
+          .setDescription(listOfLanguages);
+        message.send({ embed });
+      },
+    },
+  ],
+  permissionLevels: [PermissionLevels.ADMIN],
+  execute: async function (message, args) {
```

Comment:
I'd like to have an interface here for type completion & co. `args: LanguageArgs`

Reply:
I am not sure how this part of the bot works. I assume you want an argument for the language which is being set which I am not sure how to do. Sorry

Reply:
You can take the `reload` command args as an example

Reply:
Something like this then?
```ts
execute: async function (message, args : SetArgs) {
```
and
```ts
interface SetArgs {
  language?:
    | "Czech"
    | "English"
}
```

Reply:
Exactly

Reply:
Alright, I have pushed this change to repository.

Reply:
Please name the interface sth. like `LanguageSetArgs` to be more precise.

Reply:
> 
> 
> Please name the interface sth. like `LanguageSetArgs` to be more precise.

Done